### ISSUE TITLE
Latte: Use-Case and Assistive-Service Driven Automated Accessibility Testing Framework for Android

### ISSUE THREAD

**Issue by Wotipati** [Wotipati/Blind_Accessibility_Papers#402]
### Links
- Paper: https://dl.acm.org/doi/10.1145/3411764.3445455

### Abstract
For 15% of the world population with disabilities, accessibility is arguably the most critical software quality attribute. The ever-growing reliance of users with disability on mobile apps further underscores the need for accessible software in this domain. Existing automated accessibility assessment techniques primarily aim to detect violations of predefined guidelines, thereby produce a massive amount of accessibility warnings that often overlook the way software is actually used by users with disability. This paper presents a novel, high-fidelity form of accessibility testing for Android apps, called Latte, that automatically reuses tests written to evaluate an app’s functional correctness to assess its accessibility as well. Latte first extracts the use case corresponding to each test, and then executes each use case in the way disabled users would, i.e., using assistive services. Our empirical evaluation on real-world Android apps demonstrates Latte’s effectiveness in detecting substantially more useful defects than prior techniques.

<img width="859" alt="ScreenShot 2021-07-03 1 10 51" src="https://user-images.githubusercontent.com/26996041/124302101-adf72080-db9b-11eb-8c8f-cebf739a5faf.png">
<img width="794" alt="ScreenShot 2021-07-03 1 11 00" src="https://user-images.githubusercontent.com/26996041/124302103-af284d80-db9b-11eb-8b8a-cfb5de8c74d7.png">
<img width="793" alt="ScreenShot 2021-07-03 1 11 07" src="https://user-images.githubusercontent.com/26996041/124302105-af284d80-db9b-11eb-9075-d2010b758d55.png">

- モバイルアプリのアクセシビリティを評価するための手法を開発
